src/fetchers/streak-fetcher.js

```diff
@@ -0,0 +1,123 @@
+import retryer from "../common/retryer.js";
+import { MissingParamError, request } from "../common/utils.js";
+
+const QUERY = `
+query streakInfo($userName:String!) { 
+  user(login: $userName){
+    name
+    contributionsCollection {
+      contributionCalendar {
```

Comment:
Remember that this will only fetch streak information for the current year. While you could retrieve longer streaks through pagination, it's worth noting that, as explained in detail [here](https://github.com/anuraghazra/github-readme-stats/pull/1691#issuecomment-1484863716), doing so is impractical on Vercel. This is because it would deplete our Public GraphQL points more rapidly. However, we do have a solution in mind—we can introduce this feature as an environment variable. This way, it can be utilized on private Vercel instances and the upcoming [GitHub action](#2179). I've also suggested a similar approach in [this GitHub issue](https://github.com/anuraghazra/github-readme-stats/issues/2282).